data/mods/Aftershock/spells/psionics/electrokinesis_concentration_eocs.json

```diff
@@ -230,7 +230,7 @@
       { "math": [ "u_afs_electrokin_personal_battery_count = 0" ] },
       {
         "u_run_inv_eocs": "all",
-        "search_data": [ { "uses_energy": true } ],
+        "search_data": [ { "uses_energy": true, "excluded_flags": [ "INTEGRATED", "PSEUDO", "USES_BIONIC_POWER" ] } ],
```

Comment:
What is the reasoning for excluding the integrated flag?  There are integrated armors, such as the internal battery compartment, that it makes sense to be charged.  I can see the other flags being used to prevent errors, but are there any items that have the integrated flag that are causing errors?

Reply:
Actually, the very first item that was bugged for me was integrated welder from multitool bionics. So yes, there are many of them - welder, multimeter, electrokit, mp3. integrated ar. Also, `search_data` check is strictly AND, so I'm limited in fixing. Repairing 5 items and breaking one (and it does not bug out into error screen, just refuses to recharge) is a lot better than current situation.

Alternatively, someone can rewrite `item::uses_energy()` or write some special one strictly for EoC checks, but this is currently slightly beyond my C++ skills.

Reply:
Yes, but all of those items also have the "USES_BIONIC_POWER" flag as well, which means they would still be excluded even with removing the integrated flag from the search data, correct?

And as another note, have you tested this on the most recent build?  I've tried testing this myself so I can give a more accurate comment, but when I add the excluded flags to the eoc my game will instantly crash upon using the power

Reply:
Hmm. I just tested it by debug-installing all bionics and turning it on and I'm not getting a crash

Reply:
Strange that my game is crashing then.  Just another thing to look into I guess

Reply:
I've tested it again and yes, it looks like that only having USES_BIONIC_POWER is enough even with nomad self-cooling armors.